application/src/main/java/org/togetherjava/tjbot/commands/moderation/ReportCommand.java

```diff
@@ -0,0 +1,179 @@
+package org.togetherjava.tjbot.commands.moderation;
+
+import com.github.benmanes.caffeine.cache.Cache;
+import com.github.benmanes.caffeine.cache.Caffeine;
+import net.dv8tion.jda.api.EmbedBuilder;
+import net.dv8tion.jda.api.JDA;
+import net.dv8tion.jda.api.entities.MessageEmbed;
+import net.dv8tion.jda.api.entities.channel.concrete.TextChannel;
+import net.dv8tion.jda.api.events.interaction.ModalInteractionEvent;
+import net.dv8tion.jda.api.events.interaction.command.MessageContextInteractionEvent;
+import net.dv8tion.jda.api.interactions.InteractionHook;
+import net.dv8tion.jda.api.interactions.commands.build.Commands;
+import net.dv8tion.jda.api.interactions.components.Modal;
+import net.dv8tion.jda.api.interactions.components.text.TextInput;
+import net.dv8tion.jda.api.interactions.components.text.TextInputStyle;
+import net.dv8tion.jda.api.requests.restaction.MessageCreateAction;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import org.togetherjava.tjbot.commands.BotCommandAdapter;
+import org.togetherjava.tjbot.commands.CommandVisibility;
+import org.togetherjava.tjbot.commands.MessageContextCommand;
+import org.togetherjava.tjbot.config.Config;
+
+import java.awt.*;
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Predicate;
+import java.util.regex.Pattern;
+
+
+/**
+ * Implements the /report command, which allows users to contact a moderator within the server which
+ * forwards messages to moderators in a dedicated channel given by
+ * {@link Config#getModMailChannelPattern()}.
+ *
+ * The /report command would give us necessary information about the offending user, and specific
+ * message that was offensive.
+ */
+public final class ReportCommand extends BotCommandAdapter implements MessageContextCommand {
+
+    private static final Logger logger = LoggerFactory.getLogger(ReportCommand.class);
+    private static final String COMMAND_NAME = "report";
+    private static final String REPORT_REASON = "reportReason";
+    private static final int COOLDOWN_DURATION_VALUE = 3;
```

Comment:
> Also instead of writing `Done` every time you can click the `Resolve conversation` button

Zabuzard doesn't like when someone else resolves his convos

Reply:
Zabuzard told me to do it this way

Reply:
[No?](https://github.com/Together-Java/TJ-Bot/pull/653#discussion_r1004075382) He even chose the same name

Reply:
i did say that. thing is, it should clarify what it actually is.

Reply:
so I am not sure now , should I add input to it? or?

Reply:
my argumentation: right now, one would think that it is actually the report reason itself. like the message. but thats incorrect, its the name/id of the input field in the modal.

this has to be clarified with the name. but maybe _INPUT is also not ideal. maybe `REPORT_REASON_INPUT_ID`?

Reply:
Sure lets go with that, for now